### PR TITLE
osd/OSD: add meta crush_device_class if read_meta fail

### PR DESCRIPTION
**Pull request by gmayyyha**
Signed-off-by: Yanhu Cao <gmayyyha@gmail.com>

### CODE REVIEW

**Comment by gmayyyha** [comment]
retest this please

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -3535,6 +3535,9 @@ int OSD::update_crush_device_class()
   int r = store->read_meta("crush_device_class", &device_class);
   if (r < 0 || device_class.empty()) {
     device_class = store->get_default_device_class();
+    r = store->write_meta("crush_device_class", device_class);
```
Comment (xiexingguo) [inline src/osd/OSD.cc]: The "crush_device_class" file is an old fashion used by ceph-disk to specify the device class for an OSD during the preparing stage. Actually I can see no benefit of storing the device classes locally...
User might reset an OSD's device class to anything he/she likes(e,g,, a pool name) though mon-CLI and then the whole  update_crush_device_class() procedure should be equivalent to a noop...

Comment (gmayyyha) [inline src/osd/OSD.cc]: @xiexingguo close this?

Comment (xiexingguo) [inline src/osd/OSD.cc]: suit yourself :-)

**Comment by liewegas** [comment]
yeah i think having the local file is more likely to lead to confusion (will get out of sync w/ mon)

**Pull request closed without merging**